src/handlers/notify_zulip.rs

```diff
@@ -236,6 +244,23 @@ pub(super) async fn handle_input<'a>(
     Ok(())
 }
 
+async fn get_zulip_ids(ctx: &Context, recipients: &[User]) -> String {
```

Comment:
this code here is duplicating what we have in `src/zulip.rs` so a better way would be to refactor that module and expose functions. Maybe another time.

Reply:
fixed (with not so good-looking code, open to suggestions)